### PR TITLE
Update client-go install instructions

### PR DESCRIPTION
**Pull request by liggitt**
#### What type of PR is this?

/kind documentation

#### What this PR does / why we need it:

Simplifies and prioritizes client-go install instructions. Adds a troubleshooting section with common errors and the mechanism for resolving them.

/cc @sttts @nikhita 

```release-note
NONE
```

### CODE REVIEW

**Comment by liggitt** [comment]
also updated README update in https://github.com/kubernetes/client-go/compare/master...liggitt:update-readme?expand=1

File: staging/src/k8s.io/client-go/INSTALL.md [hunk 1]
```diff
@@ -17,65 +15,86 @@ The next time you `go build`, `go test`, or `go run` your project,
...
+go get k8s.io/client-go@kubernetes-1.16.3
 ```
 
-Ensure your project has a `go.mod` file defined at the root of your project.
-If you do not already have one, `go mod init` will create one for you:
+You can now import and use the `k8s.io/client-go` APIs in your project.
+The next time you `go build`, `go test`, or `go run` your project,
+`k8s.io/client-go` and its dependencies will be downloaded (if needed),
+and detailed dependency version info will be added to your `go.mod` file
+(or you can also run `go mod tidy` to do this directly).
```
Comment (nikhita) [inline staging/src/k8s.io/client-go/INSTALL.md:41]: nit: this text is under both `Using the latest version` and `Using a specific version`. Is that intentional/can we remove one of them?

**Review by nikhita — COMMENTED** [review]
Commented with a non-blocking nit (wasn't sure if it's intentional)

/lgtm

/hold
please cancel the hold when you are ready

File: staging/src/k8s.io/client-go/INSTALL.md (hunk 1 above)
Comment (liggitt) [reply staging/src/k8s.io/client-go/INSTALL.md:41]: interesting... it was in two places in the doc previously as well... I think I'll merge and let a deduplication be a follow-up.

File: staging/src/k8s.io/client-go/INSTALL.md [hunk 2]
```diff
@@ -17,65 +15,86 @@ The next time you `go build`, `go test`, or `go run` your project,
...
-Reasons why you might need to use a dependency management system:
-* You use a dependency that client-go also uses, and don't want two copies of
-  the dependency compiled into your application. For some dependencies with
-  singletons or global inits (e.g. `glog`) this wouldn't even compile...
-* You want to lock in a particular version (so you don't have to change your
-  code every time we change a public interface).
-* You want your install to be reproducible. For example, for your CI system or
-  for new team members.
+- If you are using Kubernetes versions >= `v1.17.0`, use a corresponding `v0.x.y` tag.
+  For example, `k8s.io/client-go@v0.20.4` corresponds to Kubernetes `v1.20.4`:
```
Comment (pacoxu) [inline staging/src/k8s.io/client-go/INSTALL.md:24]: Need we add an example for pre-released version like `v0.21.0-beta.0`?

**Pull request merged**